### PR TITLE
Add black check into CI and badge in README.md

### PR DESCRIPTION
**Pull request by refeed**
Closes https://github.com/zulip/zulip-archive/issues/60

### CODE REVIEW

**Comment by refeed** [comment]
FYI, github action result: https://github.com/refeed/zulip-archive/runs/7261419418?check_suite_focus=true

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+jobs:
+  black:
+    runs-on: ubuntu-latest
+    name: Lint - Code formatting (black)
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
```
Comment (rht) [inline .github/workflows/lint.yml:13]: 3.6 has been EOL for a while.

**Comment by rht** [comment]
I know that ZT does it the way you do it in this PR, but PSF has an official GA for Black: https://black.readthedocs.io/en/stable/integrations/github_actions.html.
^ @neiljp you might want to do it in the ZT repo as well.

I'd say using their GA is faster, because Black is pre-baked in the Docker image, and so there is no need for `pip install` and so on.

**Comment by refeed** [comment]
Whoa, yup, I did copy it from zt. Gonna update the PR soon

**Review by neiljp — COMMENTED** [review]
@refeed I just gave this a quick look-over since @rht tagged me.

In addition to the inline points you might wish to look at the 'on:' and 'concurrency:' parts of the zulip/zulip actions and in zulip/zulip-terminal#1233, though they're less important for here I suspect.

Regarding the choice to use the built-in action, in the case of no infrastructure then it's a great choice, like here at least for now. I have concerns at least regarding synchronizing the version, options and so on between it and what is used in other tools locally - the action may read options from the repo, but that's unclear? For ZT specifically we may be able to get closer by not running the full setup install, but that's fine tuning given how long other tasks take in CI.

File: lib/date_helper.py [hunk 2]
```diff
@@ -3,5 +3,5 @@
 # I don't love this format, feel free to change (I just
 # extracted it from prior code).
 def format_date1(ts):
-    """ Nov 05 2019 at 02:51"""
+    """Nov 05 2019 at 02:51"""
```
[the remark this answers is not in the document]
Comment (refeed) [reply lib/date_helper.py:6]: Thanks, that actually is modification from `black`

[the remark this answers is not in the document]
Comment (neiljp) [reply lib/date_helper.py:6]: Ah :) I expected there to be more changes than this I think :)

[the remark this answers is not in the document]
Comment (refeed) [reply lib/date_helper.py:6]: yeah, actually there are some commits from @rht that make the codebase compliant to black, like https://github.com/zulip/zulip-archive/commit/55ce5450ea4224c39c8b784a4e03b365b51a3839

[the remark this answers is not in the document]
Comment (neiljp) [reply lib/date_helper.py:6]: In any case, this *looks* unrelated, so mentioning it in the commit text would be useful.
(it doesn't fit into the two points you list in the commit title)

[the remark this answers is not in the document]
Comment (rht) [reply lib/date_helper.py:6]: Or you can apply Black to the repo as a preparatory commit.

[the remark this answers is not in the document]
Comment (refeed) [reply lib/date_helper.py:6]: alright

**Pull request merged**